### PR TITLE
salt.loader: add error logging when whitelist lookup fails

### PR DESCRIPTION
**Pull request by terminalmage**
Rather than just raising a bare KeyError with no additional information, this raises a proper KeyError with the key that failed to be looked up. It also logs the key that failed to load, as well as the whitelist, to aid in troubleshooting.

Additionally, this resolves failed lazydict lookups in the tests due to improperly-set whitelists.

### CODE REVIEW

File: salt/loader.py [hunk 1]
```diff
@@ -1629,7 +1629,11 @@ def _load(self, key):
                 return True
             # if the modulename isn't in the whitelist, don't bother
             if self.whitelist and mod_name not in self.whitelist:
-                raise KeyError
+                log.error(
+                    'Failed to load key %s because its module (%s) is not in '
```
Comment (isbm) [inline salt/loader.py]: If someone reading the log won't start panicking about SSH pr SSL keys... Can we rephrase this to something else?

**Review by isbm — CHANGES_REQUESTED** [review]

**Comment by rallytime** [comment]
@terminalmage The dockermod tests are failing on this change. Can you take a look?

https://jenkins.saltstack.com/job/PR/job/salt-pr-linode-cent7-py3/5009/

File: salt/loader.py (hunk 1 above)
Comment (terminalmage) [reply salt/loader.py]: Sure, that is a good point.

Comment (terminalmage) [reply salt/loader.py]: Done.

**Comment by terminalmage** [comment]
@rallytime Should be fixed now. I though I ran all the dockermod unit tests after editing the whitelist, but it looks like I only ran a single test.

**Review by cachedout — APPROVED** [review]

**Review by rallytime — APPROVED** [review]
